tests/kernel/device/kernel.device.pm fails to build on TI platforms

`./scripts/twister -s tests/kernel/device/kernel.device.pm  -p cc26x2r1_launchxl`

Gets:

```
/home/galak/git/zephyr/soc/arm/ti_simplelink/cc13x2_cc26x2/power.c:60:1: error:expected identifier or '(' before '{' token
   60 | {
      | ^
/home/galak/git/zephyr/soc/arm/ti_simplelink/cc13x2_cc26x2/power.c:122:1: error: expected identifier or '(' before '{' token
  122 | {
      | ^
/home/galak/git/zephyr/soc/arm/ti_simplelink/cc13x2_cc26x2/power.c:188:1: error: expected identifier or '(' before '{' token
  188 | {
      | ^
/home/galak/git/zephyr/soc/arm/ti_simplelink/cc13x2_cc26x2/power.c:202:1: error: expected identifier or '(' before '{' token
  202 | {
      | ^
In file included from /home/galak/git/zephyr/include/toolchain/gcc.h:67,
                 from /home/galak/git/zephyr/include/toolchain.h:50,
                 from /home/galak/git/zephyr/include/kernel_includes.h:19,
                 from /home/galak/git/zephyr/include/kernel.h:17,
                 from /home/galak/git/zephyr/include/zephyr.h:18,
                 from /home/galak/git/zephyr/soc/arm/ti_simplelink/cc13x2_cc26x2/power.c:6:
/home/galak/git/zephyr/include/pm/pm.h:209:38: error: expected identifier or '(' before numeric constant
  209 | #define pm_constraint_get(pm_state) (true)
      |                                      ^~~~
/home/galak/git/zephyr/soc/arm/ti_simplelink/cc13x2_cc26x2/power.c:215:6: note:in expansion of macro 'pm_constraint_get'
  215 | bool pm_constraint_get(enum pm_state state)
      |      ^~~~~~~~~~~~~~~~~
make[2]: *** [zephyr/CMakeFiles/zephyr.dir/build.make:440: zephyr/CMakeFiles/zephyr.dir/soc/arm/ti_simplelink/cc13x2_cc26x2/power.c.obj] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:2564: zephyr/CMakeFiles/zephyr.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

--- a/soc/arm/ti_simplelink/cc13x2_cc26x2/power.c
+++ b/soc/arm/ti_simplelink/cc13x2_cc26x2/power.c
@@ -48,6 +48,7 @@ const PowerCC26X2_Config PowerCC26X2_config = {
 
 extern PowerCC26X2_ModuleState PowerCC26X2_module;
 
+#ifdef CONFIG_PM
 /*
  * Power state mapping:
  * PM_STATE_SUSPEND_TO_IDLE: Idle
@@ -126,6 +127,7 @@ __weak void pm_power_state_exit_post_ops(struct pm_state_info info)
 	 */
 	CPUcpsie();
 }
+#endif /* CONFIG_PM */
 
 /* Initialize TI Power module */
 static int power_initialize(const struct device *dev)
@@ -182,6 +184,7 @@ void PowerCC26XX_schedulerRestore(void)
 	 */
 }
 
+#ifdef CONFIG_PM
 /* Constraint API hooks */
 
 void pm_constraint_set(enum pm_state state)
@@ -231,6 +234,7 @@ bool pm_constraint_get(enum pm_state state)
 
 	return ret;
 }
+#endif /* CONFIG_PM */
 
 SYS_INIT(power_initialize, POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT);
 SYS_INIT(unlatch_pins, POST_KERNEL, CONFIG_APPLICATION_INIT_PRIORITY);

